x/vendorinfo/handler_test.go

```diff
@@ -15,280 +15,278 @@
+	require.NoError(t, err)
+
+	// check
+	require.Equal(t, msgCreateVendorInfo.VendorID, receivedVendorInfo.VendorID)
+	require.Equal(t, msgCreateVendorInfo.CompanyLegalName, receivedVendorInfo.CompanyLegalName)
+	require.Equal(t, msgCreateVendorInfo.CompanyPrefferedName, receivedVendorInfo.CompanyPrefferedName)
+	require.Equal(t, msgCreateVendorInfo.Creator, receivedVendorInfo.Creator)
+	require.Equal(t, msgCreateVendorInfo.VendorLandingPageURL, receivedVendorInfo.VendorLandingPageURL)
+	require.Equal(t, msgCreateVendorInfo.VendorName, receivedVendorInfo.VendorName)
+}
+
+func queryVendorInfo(
+	setup TestSetup,
+	vid int32,
+) (*types.VendorInfo, error) {
+	req := &types.QueryGetVendorInfoRequest{
+		VendorID: vid,
+	}
+
+	resp, err := setup.Keeper.VendorInfo(setup.Wctx, req)
+	if err != nil {
+		require.Nil(setup.T, resp)
+		return nil, err
+	}
+
+	require.NotNil(setup.T, resp)
+	return &resp.VendorInfo, nil
+}
+
+func TestHandler_UpdateVendorInfo(t *testing.T) {
+	setup := Setup(t)
+
+	// try update not present vendorinfo
+	msgUpdateVendorInfo := NewMsgUpdateVendorInfo(setup.Vendor)
+	_, err := setup.Handler(setup.Ctx, msgUpdateVendorInfo)
+	println(err.Error())
+	require.Error(t, err)
+
+	// add new vendorinfo
+	msgCreateVendorInfo := NewMsgCreateVendorInfo(setup.Vendor)
+	_, err = setup.Handler(setup.Ctx, msgCreateVendorInfo)
+	require.NoError(t, err)
+
+	// update existing vendorinfo
+	_, err = setup.Handler(setup.Ctx, msgUpdateVendorInfo)
+	require.NoError(t, err)
+
+	// query updated vendorinfo
+	receivedVendorInfo, err := queryVendorInfo(setup, msgUpdateVendorInfo.VendorID)
+	require.NoError(t, err)
+
+	// check
+	require.Equal(t, msgUpdateVendorInfo.VendorID, receivedVendorInfo.VendorID)
+	require.Equal(t, msgUpdateVendorInfo.CompanyLegalName, receivedVendorInfo.CompanyLegalName)
+	require.Equal(t, msgUpdateVendorInfo.CompanyPrefferedName, receivedVendorInfo.CompanyPrefferedName)
+	require.Equal(t, msgUpdateVendorInfo.Creator, receivedVendorInfo.Creator)
+	require.Equal(t, msgUpdateVendorInfo.VendorLandingPageURL, receivedVendorInfo.VendorLandingPageURL)
+	require.Equal(t, msgUpdateVendorInfo.VendorName, receivedVendorInfo.VendorName)
+}
+
+func TestHandler_OnlyOwnerCanUpdateVendorInfo(t *testing.T) {
+	setup := Setup(t)
+
+	// add new vendorinfo
+	msgCreateVendorInfo := NewMsgCreateVendorInfo(setup.Vendor)
+	_, err := setup.Handler(setup.Ctx, msgCreateVendorInfo)
+	require.NoError(t, err)
+
+	for _, role := range []dclauthtypes.AccountRole{
+		dclauthtypes.TestHouse,
+		dclauthtypes.CertificationCenter,
+		dclauthtypes.Trustee,
+		dclauthtypes.NodeAdmin,
+	} {
+		accAddress := GenerateAccAddress()
+		setup.AddAccount(accAddress, []dclauthtypes.AccountRole{role}, setup.VendorID)
+
+		// update existing vendorinfo by user without Vendor role
+		msgUpdateVendorInfo := NewMsgUpdateVendorInfo(accAddress)
+		_, err = setup.Handler(setup.Ctx, msgUpdateVendorInfo)
+		require.NoError(t, err)
```

Comment:
So, a use without a Vendor role can update existing VendorInfo.... Is it expected? I think no.

Reply:
fixed